src/_pytest/main.py

```diff
@@ -391,22 +412,30 @@ class Session(nodes.FSCollector):
     _setupstate = None  # type: SetupState
     # Set on the session by fixtures.pytest_sessionstart.
     _fixturemanager = None  # type: FixtureManager
+    exitstatus = None  # type: Union[int, ExitCode]
 
     def __init__(self, config: Config) -> None:
         nodes.FSCollector.__init__(
             self, config.rootdir, parent=None, config=config, session=self, nodeid=""
         )
         self.testsfailed = 0
         self.testscollected = 0
-        self.shouldstop = False
-        self.shouldfail = False
+        self.shouldstop = False  # type: Union[bool, str]
+        self.shouldfail = False  # type: Union[bool, str]
         self.trace = config.trace.root.get("collection")
         self._norecursepatterns = config.getini("norecursedirs")
         self.startdir = config.invocation_dir
         self._initialpaths = frozenset()  # type: FrozenSet[py.path.local]
 
         # Keep track of any collected nodes in here, so we don't duplicate fixtures
-        self._node_cache = {}  # type: Dict[str, List[Node]]
+        self._node_cache1 = {}  # type: Dict[py.path.local, Sequence[nodes.Collector]]
+        self._node_cache2 = (
```

Comment:
(We discussed this in another thread, but worth mentioning here)

I still think it is not a good approach to change/create attributes in bugfix releases, so this should go into `features`.

Reply:
Oh and this will also be a headache to merge with `features` later, because those names have changed there already. 😞

Reply:
@nicoddemus JFI: this is not really part of this PR (just includes it (https://github.com/pytest-dev/pytest/pull/6555) for now)
As for reviewing, you can look at the commits this is adding.